engine/src/main/java/io/zeebe/engine/state/mutable/MutableWorkflowState.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.engine.state.mutable;
+
+import io.zeebe.engine.state.immutable.WorkflowState;
+import io.zeebe.protocol.impl.record.value.deployment.DeploymentRecord;
+import org.agrona.DirectBuffer;
+
+public interface MutableWorkflowState extends WorkflowState {
+
+  void putDeployment(DeploymentRecord deploymentRecord);
+
+  void putLatestVersionDigest(DirectBuffer processId, DirectBuffer digest);
+
+  int getNextWorkflowVersion(String bpmnProcessId);
```

Comment:
Let's rename this to `nextWorkflowVersion` to make it more clear that this is not a getter without a side-effect